### PR TITLE
Enable lint: UnusedResources

### LINKED ISSUE #7571: Turn on Lint Error: UnusedResources
Flagged this morning - after a check, it turns out that we have ~~90 strings which are unused on CrowdIn~~ 90 unused items in the project

This will help us keep CrowdIn free of useless strings and reduce the workload of our translators, and will stop it from happening again in the future.

One of the problems with removed strings is that we don't know whether they were removed due to error, or due to deprecation. Having an active lint check will enforce this choice on the author of the PR, who will know best.

Aims:
* One large commit with all deprecated strings removed - submitted as a large PR and squash-merged
* Individual commits fixing individual issues (strings to be used, or fixing errors).

### PR DESCRIPTION
**Pull request by david-allison-1**
Please review this commit-by-commit

## Purpose / Description
`UnusedResources` will be a useful lint check - as it means unused files and strings will either be flagged, or explicitly removed closest to the context of why they were removed

## Fixes
Fixes #7571 

## Approach
Remove strings
Removed resources in previous PRs linked to #7571 
Ignored a couple of files which were OK

## How Has This Been Tested?

`gradlew AnkiDroid:lintRelease` passes. Ran on API 16 emulator

## Learning
`<ignore regex>` and `<ignore path>` are really useful

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] Strings resources: All strings added as resources are unique or contain a comment for seemingly duplicate strings to explain the difference between both resources

### CODE REVIEW

**Comment by david-allison-1** [comment]
Originally, I felt this would be best to squash-merge, I think that was the wrong decision and it'd be better to rebase this PR to allow specific reverts if any change was a mistake.

Maintainer's choice

File: lint-release.xml [hunk 1]
```diff
@@ -29,6 +29,10 @@
         <!-- Ignore as they'll be used for #7526 -->
         <ignore regexp="content_picture\.png|device_access_mic\.png" />
         <ignore path="res/values/12-dont-translate.xml" />
+        <!-- Ignore all translations except values - CrowdIn will handle deletions -->
+        <!-- We only need to handle -af and all others are implicitly ignored -->
+        <!-- tested and this does not block a string in /values/ from triggering lint -->
+        <ignore path="res/values-af/" />
```
Comment (mikehardy) [inline lint-release.xml:35]: why is values-af special?

**Review by mikehardy — APPROVED** [review]
I like this - I'm just curious about values-af

**Comment by david-allison-1** [comment]
> I like this - I'm just curious about values-af

I'll post the validation errors when I'm back at my PC, let's hold off on this until then

**Comment by david-allison-1** [comment]
```

  The first 3 errors (out of 24) were:
  C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\res\values-af\02-strings.xml:46: Error: The resource R.string.CardEditorLaterMessage appears to be unused [UnusedResources]
      <string name="CardEditorLaterMessage">Inligting vir later gestoor</string>
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\res\values-af\02-strings.xml:58: Error: The resource R.string.sdcard_missing_message appears to be unused [UnusedResources]
      <string name="sdcard_missing_message">Switch off USB storage to access your decks</string>
              ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  C:\GitHub\Anki-Android-Git\Anki-Android\AnkiDroid\src\main\res\values-af\02-strings.xml:120: Error: The resource R.string.add appears to be unused [UnusedResources]
      <string name="add">Voeg by</string>
              ~~~~~~~~~~

```

**Comment by mikehardy** [comment]
Fascinating. So basically, if you delete a resource in values/ *and* you have this lint on, then you will always have lint flag some other folder imported from crowdin until you do an i18n sync that purges out the derivative folders.

But, if you specify the first directory, then the others are skipped!? Is there a way to skip values-* ? (is that even advisable...)

**Comment by david-allison-1** [comment]
> Fascinating. So basically, if you delete a resource in values/ _and_ you have this lint on, then you will always have lint flag some other folder imported from crowdin until you do an i18n sync that purges out the derivative folders.
> 
> But, if you specify the first directory, then the others are skipped!? Is there a way to skip values-* ? (is that even advisable...)

I tried, and couldn't get it to work - it seems that `values-af` is functionally equivalent to `values-*`

**Comment by mikehardy** [comment]
...and so it goes! Software never fails to surprise.

**Pull request merged**